skopt/tests/test_tree_opt.py

```diff
@@ -0,0 +1,102 @@
+from functools import partial
+
+import numpy as np
+
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_almost_equal
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_less
+from sklearn.utils.testing import assert_raise_message
+
+from skopt.benchmarks import bench1
+from skopt.benchmarks import bench2
+from skopt.benchmarks import bench3
+from skopt.benchmarks import branin
+from skopt.benchmarks import hart6
+from skopt.tree_opt import gbrt_minimize
+from skopt.tree_opt import forest_minimize
+
+
+def check_no_iterations(minimizer):
+    assert_raise_message(ValueError, "at least one iteration",
+                         minimizer,
+                         branin, [(-5.0, 10.0), (0.0, 15.0)], maxiter=0,
+                         random_state=1)
+
+    assert_raise_message(ValueError, "at least one starting point",
+                         minimizer,
+                         branin, [(-5.0, 10.0), (0.0, 15.0)], n_start=0,
+                         maxiter=2, random_state=1)
+
+
+def test_no_iterations():
+    yield (check_no_iterations, gbrt_minimize)
+    yield (check_no_iterations, partial(forest_minimize, base_estimator='et'))
+    yield (check_no_iterations, partial(forest_minimize, base_estimator='dt'))
+    yield (check_no_iterations, forest_minimize)
+
+
+def test_one_iteration():
+    result = gbrt_minimize(branin, [(-5.0, 10.0), (0.0, 15.0)],
```

Comment:
Should we not raise an error asking the user to either set `maxiter >= n_start`?

IMO, we should not implicitly change the default value of `n_start=10` or in other words the user should know what is happening.